server/consumer.go

```diff
@@ -2070,8 +2075,10 @@ func (o *consumer) getNextMsg() (subj string, hdr, msg []byte, seq uint64, dc ui
 // Lock should be held.
 func (o *consumer) forceExpireFirstWaiting() *waitingRequest {
 	// FIXME(dlc) - Should we do advisory here as well?
-	wr := o.waiting.pop()
-	if wr == nil {
+	wr := o.waiting.peek()
+	if wr != nil {
+		o.waiting.done()
```

Comment:
I don't feel confident to review this change, but what strikes me is that here you are by-passing the "pop()" which does reduce wr.n, and only remove when count is <= 0. Not sure what this means really, so again, I am not the best to review this.